source/common/ssl/ssl_socket.cc

```diff
@@ -17,6 +17,38 @@ using Envoy::Network::PostIoAction;
 namespace Envoy {
 namespace Ssl {
 
+namespace {
+
+// This SslSocket will be used when SSL secret is not fetched from SDS server.
```

Comment:
Do we really need this? What causes us to require an instantiated connection? Can't we cause the control flow to return in such a way that there is no socket and we just fail whatever we are doing?

Reply:
Our first attempt was to return nullptr in socket create when ssl_context is not ready (SDS client failed to get secret).  This for for ListenerImpl,  but did not work for ClusterImpl.  Look at the ClusterImpl codes, most of them not expect nullptr socket nor nullptr connection.  It will require a lot of code changes to make these code to handle null connection.

For less code change, we decided to return such dummy socket which would just reset the connect.

Reply:
I don't have an opinion on whether this is the best course of action or not. @htuch hopefully can have a look. In general I would prefer that we don't do this but if it's the best way I can accept that.